ingest/src/withConfig.ts

```diff
@@ -0,0 +1,180 @@
+import yargs from "yargs";
+import Path from "path";
+import { logger } from "chat-core";
+import { Config } from "./Config";
+
+export type LoadConfigArgs = {
+  config?: string;
+};
+
+export const loadConfig = async ({
+  config: configPathIn,
+}: LoadConfigArgs): Promise<Config> => {
+  const path = Path.resolve(
+    configPathIn === undefined ? "ingest.config.cjs" : configPathIn
+  );
+
+  const configArray = (await import(path)).default as Partial<Config>[];
+  console.log(JSON.stringify(configArray));
+  // Validate config
+  if (!Array.isArray(configArray)) {
+    throw new Error(
+      `Invalid config at ${path}: expected default exported array of Config objects`
+    );
+  }
+
+  if (configArray.length === 0) {
+    throw new Error(
+      `Invalid config at ${path}: expected at least 1 Config object in array`
+    );
+  }
+
+  // Flatten config: the fields of earlier entries in the array can be
+  // overridden by the fields of subsequent entries in the array.
```

Comment:
i don't think this is desired behavior. the way i think about it, each `Config` in the array should be independent. why do you think it should be this way?

Reply:
That's how it works in eslint's flat cascade? https://eslint.org/blog/2022/08/new-config-system-part-2/#goodbye-extends%2C-hello-flat-cascade

Reply:
i think that factor of the flat config would only apply here if both configs touch the same data source. ie both are for `snooty-docs-realm`. but if there's config1 with `snooty-docs-realm` using embeddedContentStore1`, and config2 with `snooty-docs-app-services` using `embeddedContentStore2`, that should be allowed.